infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesEnvironmentProvisioner.java

```diff
@@ -97,6 +103,7 @@ public KubernetesEnvironmentProvisionerImpl(
         CertificateProvisioner certificateProvisioner,
         SshKeysProvisioner sshKeysProvisioner,
         GitConfigProvisioner gitConfigProvisioner,
+        TrustStoreCertificateProvisioner trustStoreCertificateProvisioner,
```

Comment:
IMHO it's wrong place for this provisioner. It makes it only provisioner here that uses k8s client and that is actually creating k8s objects, which is wrong here. I see better fit to put it one level higher into `KubernetesInternalRuntime`. Or is there any good reason to have it here and actually communicate with k8s ?